src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -172,12 +174,21 @@ class JobHistoryContent extends Component {
                   ]),
                   h(PopupTrigger, {
                     position: 'bottom',
-                    content: h(MenuButton, {
-                      as: 'a',
-                      href: `${firecloudRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
-                    }, [
-                      icon('circle-arrow right'),
-                      div({ style: { marginLeft: '0.5rem' } }, ['View job details'])
+                    content: h(Fragment, [
+                      h(MenuButton, {
+                        as: 'a',
+                        target: '_blank',
+                        href: `${firecloudRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
+                      }, [
+                        icon('circle-arrow right'),
+                        div({ style: { marginLeft: '0.5rem' } }, ['View job details'])
+                      ]),
+                      collapsedStatuses(submissions[rowIndex].workflowStatuses).running && h(MenuButton, {
+                        onClick: () => this.setState({ aborting: rowIndex })
```

Comment:
Overall I like this pattern. But by using the row number as an id here, there's an (unlikely, but possible) edge case to consider: If a new job gets launched in the background, and a refresh happens while the confirmation modal is open, it could shift the index, and the person might cancel the wrong job. If we think this is worth worrying about, then we could use the job id instead of the index.